src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -32,6 +34,29 @@
     private static Bundle gCachedExtras = null;
     private static boolean gForeground = false;
 
+    @Override
+    public void initialize(CordovaInterface cordova, CordovaWebView webView) {
+        super.initialize(cordova, webView);
+        gForeground = true;
+
+        if (isAmazonDevice()) {
```

Comment:
If I turn this into `isGoogleDevice` the detection will probably still be "manufacturer is not amazon" and flipping this by turning the condition into `!isAmazonDevice` would leave with a negated `if` with an `else` clause which also is an antipattern. Maybe I'm missing your point?